plugins/node/opentelemetry-instrumentation-mongodb/src/instrumentation.ts

```diff
@@ -353,7 +353,12 @@ export class MongoDBInstrumentation extends InstrumentationBase {
           instrumentation.setPoolName(options);
           callback(err, conn);
         };
-        return original.call(this, options, patchedCallback);
+
+        const result = original.call(this, options, patchedCallback);
```

Comment:
It feels slightly messy to be passing in this `patchedCallback` that isn't going to be used.
Thoughts on a separate patch function for mongodb >=6.4.0 -- this one would just do the promise thing.

Reply:
my intention was to not split the patch logic of this function to 2 separate patches. I do agree is not optimal to crate a function that may not be used. I'll try to come with a different approach